assets/js/product.js

```diff
@@ -81,12 +81,16 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
                     });
                });
 
-                //Auto select initial product as default variation  (its possible because all our products is variations)
+                
                 var propertyMap = getVariationPropertyMap(product);
                 var key = _.first(_.keys(propertyMap));
-                $scope.checkProperty(propertyMap[key][0]);
-
-               //$scope.selectedVariation = product;
+                if (key){
+                    $scope.checkProperty(propertyMap[key][0]);
```

Comment:
Could we make sure there is no out of range exception when getting [0] in `propertyMap[key][0]`?

Reply:
Yes, I think so. Function getVariationPropertyMap use 'return groupBy...'. So if array with 'key' exists then exist one ore more items in this array.

https://github.com/VirtoCommerce/vc-theme-default/blob/e00229cee8744029609b1a9ac3d181945beb2e6e/assets/js/product.js#L106